### PR TITLE
Added OAI-PMH support

### PR DESCRIPTION
**Pull request by aldettinger**
[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

File: integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    public void startServer() {
+        server = new Server(port);
+
+        if (useHttps) {
+            HttpConfiguration https = new HttpConfiguration();
+            https.addCustomizer(new SecureRequestCustomizer());
+            SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
+
+            String keyStorePath = OaipmhTestServer.class.getResource("/jettyKS/localhost.p12").toExternalForm();
+            sslContextFactory.setKeyStorePath(keyStorePath);
```
Comment (aldettinger) [inline integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java:94]: I expect this to work on platform too.

**Review by oscerd — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -84,6 +84,7 @@
         <java.xml.ws.version>2.3.1</java.xml.ws.version>
         <jcodings.version>1.0.55</jcodings.version><!-- used by hbase -->
         <joni.version>2.1.31</joni.version><!-- used by json-validator -->
+        <jsoup.version>1.12.1</jsoup.version><!-- used by oaipmh -->
```
Comment (aldettinger) [inline pom.xml:87]: camel-oaipmh brought 1.13.1 in compile scope while quarkus-junit5 brought 1.12.1 in test scope. So I aligned on quarkus version with compile scope.

File: extensions/oaipmh/deployment/src/main/java/org/apache/camel/quarkus/component/oaipmh/deployment/OaipmhProcessor.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @BuildStep
+    ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
+        return new ExtensionSslNativeSupportBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    NativeImageResourceDirectoryBuildItem registerNativeResources() {
+        return new NativeImageResourceDirectoryBuildItem("org/joda/time/tz/data");
```
Comment (ppalaga) [inline extensions/oaipmh/deployment/src/main/java/org/apache/camel/quarkus/component/oaipmh/deployment/OaipmhProcessor.java:40]: We should consider a separate support (or Quarkiverse) extension once we have more stuff for Joda time. It is OK, as far as it is just a single item. Shouldn't we at least file a "Consider separate Joda extension" issue?

File: integration-tests/oaipmh/pom.xml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <!-- The following dependencies guarantee that this module is built after them. You can update them by running `mvn process-resources -Pformat -N` from the source tree root directory -->
```
Comment (ppalaga) [inline integration-tests/oaipmh/pom.xml:141]: I wonder why this comment is twice in the file. I hope the cq-plugin is not broken?

File: integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java (hunk 1 above)
Comment (ppalaga) [reply integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java:94]: Let's see :-)

**Review by ppalaga — APPROVED** [review]
Nice work, thanks!

File: integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                }
+                entry = zis.getNextEntry();
+            }
+            RESPONSE_CACHE = Collections.unmodifiableMap(responseCache);
+        }
+        return RESPONSE_CACHE;
+    }
+
+    public void startServer() {
+        server = new Server(port);
```
Comment (jamesnetherton) [inline integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java:86]: Jetty server :cry: 

Was hoping that we'd be able to keep it out of this project and leverage the existing Camel HTTP consumers if we need web server bits for testing. It's less maintenance overhead (in theory) etc.

Happy to leave things as is for now but maybe we should try to replace it if possible at some point?

File: extensions/oaipmh/deployment/src/main/java/org/apache/camel/quarkus/component/oaipmh/deployment/OaipmhProcessor.java (hunk 3 above)
Comment (aldettinger) [reply extensions/oaipmh/deployment/src/main/java/org/apache/camel/quarkus/component/oaipmh/deployment/OaipmhProcessor.java:40]: Filed https://github.com/apache/camel-quarkus/issues/2191.

File: integration-tests/oaipmh/pom.xml (hunk 4 above)
Comment (aldettinger) [reply integration-tests/oaipmh/pom.xml]: I've regenerated that part and it looks cleaner. I don't think the cq-plugin is broken, but maybe I misled it as I have added/removed/changed order of dependencies.

File: integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java (hunk 5 above)
Comment (aldettinger) [reply integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java:86]: Indeed, let's have a look at this in https://github.com/apache/camel-quarkus/issues/2192 to check if I can achieved to find a solution that is clear and could be applied to main Camel too.
To be clear, the idea you mentioned would be to implement a test resource launching a route that consume from camel-http ?

Comment (jamesnetherton) [reply integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java:86]: Or just have some camel routes that leverage netty-http and return the responses needed by the test.

Comment (aldettinger) [reply integration-tests/oaipmh/src/test/java/org/apache/camel/quarkus/component/oaipmh/it/OaipmhTestServer.java:86]: Ok I see, thanks for sharing ideas :+1:

**Comment by aldettinger** [comment]
The build issue is unrelated, merging.

**Pull request merged**